.travis.yml

```diff
@@ -72,12 +75,61 @@ before_install:
     - tools/check_sdist.py
 
 install:
-    - section build
     - python setup.py develop
-    - section_end build
+    # Matplotlib settings - do not show figures during doc examples
+    # Where is the matplotlib home on a mac?
+    - |
+      if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+        export MPL_DIR=${HOME}/.matplotlib
+      else
+        export MPL_DIR=${HOME}/.config/matplotlib
+      fi
+    - mkdir -p ${MPL_DIR}
+    - touch ${MPL_DIR}/matplotlibrc
+    # Install most of the optional packages
+    - |
+      if [[ "${OPTIONAL_DEPS}" == "1" ]]; then
+        pip install --retries 3 -q -r ./requirements/optional.txt $WHEELHOUSE
+      fi
+    - |
+      if [[ "${WITH_PYAMG}" == "1" ]]; then
+        pip install --retries 3 -q pyamg
+      fi
+    - source tools/travis/install_qt.sh
 
-script: tools/travis/script.sh
+script:
+    - export PY=${TRAVIS_PYTHON_VERSION}
+    - flake8 --exit-zero --exclude=test_*,six.py skimage doc/examples viewer_examples
+    # run tests again with optional dependencies to get more coverage
+    - |
+      if [[ "$OPTIONAL_DEPS" == "1" ]]; then
+        export TEST_ARGS="${TEST_ARGS} --cov=skimage"
+      fi
+    # Show what's installed
+    - pip list
+    - pytest ${TEST_ARGS} skimage
+    # Run example applications
+    - echo Build or run examples
+    - |
+      if [[ "${BUILD_DOCS}" == "1" ]]; then
+        # requirements/docs.txt fails on Travis OSX
+        pip install --retries 3 -q -r ./requirements/docs.txt
+        export SPHINXCACHE=${HOME}/.cache/sphinx; make html
+      elif [[ "${TEST_EXAMPLES}" != "0" ]]; then
+        pip install --retries 3 -q -r ./requirements/docs.txt
+        cp $MPL_DIR/matplotlibrc $MPL_DIR/matplotlibrc_backup
+        echo 'backend : Template' > $MPL_DIR/matplotlibrc
+        for f in doc/examples/*/*.py; do
+            python "${f}"
+            if [ $? -ne 0 ]; then
+                exit 1
+            fi
+        done
+        mv $MPL_DIR/matplotlibrc_backup $MPL_DIR/matplotlibrc
+      fi
 
 after_success:
     - codecov
+    # Prepare.release
+    - doc/release/contribs.py HEAD~10
```

Comment:
What's this for?

Reply:
Was in the script

https://github.com/scikit-image/scikit-image/blob/51f598aaedc73ef180913c670d2c20a8032aaf1e/tools/travis/script.sh#L121